photon-client/src/main/scala/com/linkedin/photon/ml/data/avro/AvroDataWriter.scala

```diff
@@ -93,6 +93,11 @@ object AvroDataWriter {
   protected[data] def getValueAsDouble(row: Row, fieldName: String): Double = {
 
     row.getAs[Any](fieldName) match {
+      case null => fieldName match {
+        case "offset" => 0.0D
+        case "weight" => 1.0D
```

Comment:
We should extract these out into constants for here and L57-58. Actually, the values should be in constants too. Also, we might want to abstract out the lookup of default values, e.g.:

```scala
    val OFFSET = "offset"
    val WEIGHT = "weight"

    val defaults = Map(
      OFFSET -> 0.0D,
      WEIGHT -> 1.0D)

    // ...
    row.getAs[Any](fieldName) match {
      case null => defaults.get(fieldName)
        .getOrElse {
          throw new IllegalArgumentException(s"Unsupported null for fieldName $fieldName") }
      // ..
```